Tests/Unit/QuietFactoryTest.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Symfony\Cmf\Bundle\MenuBundle\Tests\Unit;
+
+use Symfony\Cmf\Bundle\MenuBundle\QuietFactory;
+
+class QuietFactoryTest extends \PHPUnit_Framework_TestCase
+{
+    private $innerFactory;
+    private $logger;
+
+    protected function setUp()
+    {
+        $this->innerFactory = $this->prophesize('Knp\Menu\FactoryInterface');
+        $this->logger = $this->prophesize('Psr\Log\LoggerInterface');
+    }
+
+    public function testAllowEmptyItemsReturnsItemWithoutURL()
+    {
+        $this->innerFactory->createItem('Home', array('route' => 'not_existent'))
+            ->willThrow('Symfony\Component\Routing\Exception\RouteNotFoundException');
+
+        $this->innerFactory->createItem('Home', array())->shouldBeCalled();
```

Comment:
a willReturn and asserting the return value of ``$factory->createItem`` would be better than a ``shouldBeCalled`` assertion: the important part is not that the inner factory is called, but that the item is returned